truss/templates/server/common/errors.py

```diff
@@ -57,44 +37,120 @@ def __str__(self):
         return self.error_msg
 
 
-async def exception_handler(_, exc):
-    return JSONResponse(
-        status_code=HTTPStatus.INTERNAL_SERVER_ERROR, content={"error": str(exc)}
-    )
+class InputParsingError(ValueError):
+    pass
 
 
-async def invalid_input_handler(_, exc):
-    return JSONResponse(status_code=HTTPStatus.BAD_REQUEST, content={"error": str(exc)})
+class UserCodeError(Exception):
+    pass
 
 
-async def inference_error_handler(_, exc):
-    return JSONResponse(
-        status_code=HTTPStatus.INTERNAL_SERVER_ERROR, content={"error": str(exc)}
-    )
+def _make_baseten_error_headers(error_code: int) -> Mapping[str, str]:
+    # The source of truth for these constants is in
+    # go/beefeater/shared/error_propagated.go
+    truss_server_service_id = "04"
+    return {
+        "X-BASETEN-ERROR-SOURCE": truss_server_service_id,
+        "X-BASETEN-ERROR-CODE": str(error_code),
+    }
 
 
-async def generic_exception_handler(_, exc):
-    return JSONResponse(
-        status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
-        content={"error": f"{type(exc).__name__} : {str(exc)}"},
-    )
-
-
-async def model_not_found_handler(_, exc):
-    return JSONResponse(status_code=HTTPStatus.NOT_FOUND, content={"error": str(exc)})
+def _make_baseten_response(
+    http_status: int,
+    info: Union[str, Exception],
+    baseten_error_code: Optional[int] = None,
```

Comment:
is the plan to start passing this argument later?

Reply:
Yes, the concrete codes are still up for review with @nicolasgere .